notification.go

```diff
@@ -0,0 +1,78 @@
+package watchdog
+
+import "sync"
+
+var (
+	gcNotifeeMutex sync.Mutex
+	gcNotifees     []notifeeEntry
+
+	forcedGCNotifeeMutex sync.Mutex
+	forcedGCNotifees     []notifeeEntry
+)
+
+// RegisterNotifee registers a function that is called when a GC has happened.
+// The unregister function returned can be used to unregister this notifee.
+func RegisterNotifee(f func()) (unregister func()) {
```

Comment:
Let's call this `RegisterPostGCNotifee`, and specify that it will be called for both watchdog- and system-initiated GC in Godocs.

Reply:
Done.